1Password/connect-sdk-python
# Enhance README and move usage content to USAGE.md

This PR contains the following key changes:
- `README.md`
  - Content is arranged based on the new template we use for the README.
  - Provide a Quickstart section with two main cases: read and write an item.
  - Detailed content regarding the SDK is moved to a new file: `USAGE.md`.
- `USAGE.md`
  - Functions related to item and vault operations are presented in a friendlier way.
  - Code snippets are adjusted to match the latest version of the SDK, therefore they're more accurate.

### How to review this PR

1. Check the new README page by [clicking this link](https://github.com/1Password/connect-sdk-python/blob/d44f2df8266943c3f80b55bc25354062436610d7/README.md).
   - [ ] The new template is used properly.
   - [ ] The Get started page contains enough informations to give an overview of common things that users can do with the SDK. Make sure that it doesn't feel like there's too little content.
   - [ ] See it in action - is it useful?
2. Check the old README by [clicking this link](https://github.com/1Password/connect-sdk-python/blob/v1.3.0/README.md).
   - [ ] Content was properly migrated to the new files.
   - [ ] Is there anything that was cut out and should be brought back in? 
3. Check the `USAGE.md` page by [clicking this link](https://github.com/1Password/connect-sdk-python/blob/274453119481cf11fbe166c60b4d9732c745fa44/USAGE.md).
   - [ ] Flow of documentation is clean and clear.
   - [ ] Can easily find the content needed to achieve your goal.
   - [ ] Code snippets actually work.
